engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -280,7 +288,72 @@ private boolean usingBackendSnapshotFor(SnapshotInfo snapshotInfo) {
         return Boolean.parseBoolean(property);
     }
 
-    private void handleCreateTemplateFromSnapshot(SnapshotInfo snapshotInfo, TemplateInfo templateInfo, AsyncCompletionCallback<CopyCommandResult> callback) {
+    protected boolean needCacheStorage(DataObject srcData, DataObject destData) {
+        DataTO srcTO = srcData.getTO();
+        DataStoreTO srcStoreTO = srcTO.getDataStore();
+        DataTO destTO = destData.getTO();
+        DataStoreTO destStoreTO = destTO.getDataStore();
+
+        // both snapshot and volume are on primary datastore. No need for a cache storage as
+        // hypervisor will copy directly
+        if (srcStoreTO instanceof PrimaryDataStoreTO && destStoreTO instanceof PrimaryDataStoreTO) {
```

Comment:
Will this cause a problem for creating a template from a volume snapshot that's on primary storage?

On Jun 30, 2016, at 11:36 AM, Syed Mushtaq Ahmed <notifications@github.com<mailto:notifications@github.com>> wrote:


In engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java<https://github.com/syed/cloudstack/pull/4#discussion_r69175635>:

> @@ -280,7 +288,72 @@ private boolean usingBackendSnapshotFor(SnapshotInfo snapshotInfo) {
>          return Boolean.parseBoolean(property);
>      }
>
> -    private void handleCreateTemplateFromSnapshot(SnapshotInfo snapshotInfo, TemplateInfo templateInfo, AsyncCompletionCallback<CopyCommandResult> callback) {
> +    protected boolean needCacheStorage(DataObject srcData, DataObject destData) {
> +        DataTO srcTO = srcData.getTO();
> +        DataStoreTO srcStoreTO = srcTO.getDataStore();
> +        DataTO destTO = destData.getTO();
> +        DataStoreTO destStoreTO = destTO.getDataStore();
> +
> +        // both snapshot and volume are on primary datastore. No need for a cache storage as
> +        // hypervisor will copy directly
> +        if (srcStoreTO instanceof PrimaryDataStoreTO && destStoreTO instanceof PrimaryDataStoreTO) {


@mike-tutkowski<https://github.com/mike-tutkowski> I've made this condition more explicit. Does this look good?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/syed/cloudstack/pull/4/files/0db8a41472f683ebd51c59d8ad4de1a19aeeb27d#r69175635>, or mute the thread<https://github.com/notifications/unsubscribe/AC4SH2ZZz9R5BTwYrpvV_EBHtkGf7Ssrks5qQ_3_gaJpZM4I2gkI>.